### ISSUE TITLE
hspec-contrib-0.3 fails test suite with hspec-2.4 (retry broken?)

### ISSUE THREAD

**Issue by DanBurton** [hspec/hspec#307]
Seen on stackage build server, reproduced locally:

```
hspec-contrib-0.3.0: test (suite: spec)

Progress: 6/7
Test.Hspec.Contrib.HUnit
  fromHUnitTest
    works for a TestCase
    works for a labeled TestCase
    works for a TestCase with nested labels
    works for a flat TestList
    works for a nested TestList
Test.Hspec.Contrib.Retry
  retry test
    retry 11 times, then check the value FAILED [1]

Failures:

  test/Test/Hspec/Contrib/RetrySpec.hs:21:
  1) Test.Hspec.Contrib.Retry, retry test, retry 11 times, then check the value
       expected: 10
        but got: 0

Randomized with seed 516920774
```

How I reproduced:

```
$ stack unpack hspec-contrib-0.3
$ cd hspec-contrib-0.3
$ <<add the following stack.yaml>>
$ stack test
```

```yaml
# stack.yaml
resolver: nightly-2016-02-10
packages:
- '.'
extra-deps:
- hspec-core-2.4.1
- hspec-2.4.1
- hspec-discover-2.4.1
- hspec-expectations-0.8.2
- call-stack-0.1.0
```

**Comment by peti** [comment]
We have the same problem in NixOS:

~~~
Running 1 test suites...
Test suite spec: RUNNING...

Test.Hspec.Contrib.HUnit
  fromHUnitTest
    works for a TestCase
    works for a labeled TestCase
    works for a TestCase with nested labels
    works for a flat TestList
    works for a nested TestList
Test.Hspec.Contrib.Retry
  retry test
    retry 11 times, then check the value FAILED [1]

Failures:

  test/Test/Hspec/Contrib/RetrySpec.hs:21: 
  1) Test.Hspec.Contrib.Retry, retry test, retry 11 times, then check the value
       expected: 10
        but got: 0

Randomized with seed 1226969114

Finished in 0.0003 seconds
6 examples, 1 failure
Test suite spec: FAIL
Test suite logged to: dist/test/hspec-contrib-0.3.0-spec.log
0 of 1 test suites (0 of 1 test cases) passed.
~~~

**Comment by sol** [comment]
Fixed in version `0.4.0`.

**Comment by peti** [comment]
That version is not going to make it into Stackage LTS-8.x, though, is it?

**Comment by DanBurton** [comment]
@peti actually, it can! `hspec-contrib` was not included in `lts-8.0`, which means we can add any version of `hspec-contrib` that is compatible with the other packages in there. Just open an issue on https://github.com/fpco/lts-haskell/issues and we'll add it to the next lts build.
